### PR TITLE
[FLINK-28659][flink-java][JUnit5 Migration] Module: flink-java.

### PR DESCRIPTION
**Pull request by RocMarshal**
## What is the purpose of the change

*[JUnit5 Migration] Module: flink-java.*

## Brief change log

[JUnit5 Migration] Module: flink-java.

## Verifying this change

This change is already covered by existing tests.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / **no**)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / **no**)
  - The serializers: (yes / **no** / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / **no** / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: (yes / **no** / don't know)
  - The S3 file system connector: (yes / **no** / don't know)

## Documentation

  - Does this pull request introduce a new feature? (yes / **no**)
  - If yes, how is the feature documented? (**not applicable** / docs / JavaDocs / not documented)

### CODE REVIEW

**Comment by RocMarshal** [comment]
@flinkbot run azure

File: flink-java/src/test/java/org/apache/flink/api/java/utils/AbstractParameterToolTest.java [hunk 1]
```diff
@@ -19,463 +19,458 @@
...
+        assertThat(parameter.get("string", "0.0")).isEqualTo("∞");
+        assertThat(parameter.getUnrequestedParameters()).isEqualTo(Collections.emptySet());
     }
 
     @Test
-    public void testUnrequestedStringWithMissingValue() {
+    void testUnrequestedStringWithMissingValue() {
         AbstractParameterTool parameter = createParameterToolFromArgs(new String[] {"-string"});
-        Assert.assertEquals(createHashSet("string"), parameter.getUnrequestedParameters());
+        assertThat(parameter.getUnrequestedParameters()).isEqualTo(createHashSet("string"));
```
Comment (snuyanzin) [inline flink-java/src/test/java/org/apache/flink/api/java/utils/AbstractParameterToolTest.java:435]:
```suggestion
        assertThat(parameter.getUnrequestedParameters()).containsExactly("string");
```

File: flink-java/src/test/java/org/apache/flink/api/java/utils/AbstractParameterToolTest.java [hunk 2]
```diff
@@ -19,463 +19,458 @@
...
     }
 
     // Additional methods
 
     @Test
-    public void testUnrequestedHas() {
+    void testUnrequestedHas() {
         AbstractParameterTool parameter = createParameterToolFromArgs(new String[] {"-boolean"});
-        Assert.assertEquals(createHashSet("boolean"), parameter.getUnrequestedParameters());
+        assertThat(parameter.getUnrequestedParameters()).isEqualTo(createHashSet("boolean"));
```
Comment (snuyanzin) [inline flink-java/src/test/java/org/apache/flink/api/java/utils/AbstractParameterToolTest.java:446]:
```suggestion
        assertThat(parameter.getUnrequestedParameters()).containsExactly("boolean");
```

File: flink-java/src/test/java/org/apache/flink/api/java/utils/AbstractParameterToolTest.java [hunk 3]
```diff
@@ -19,463 +19,458 @@
...
 
         // test parameter access
-        Assert.assertTrue(parameter.has("boolean"));
-        Assert.assertEquals(Collections.emptySet(), parameter.getUnrequestedParameters());
+        assertThat(parameter.has("boolean")).isTrue();
+        assertThat(parameter.getUnrequestedParameters()).isEqualTo(Collections.emptySet());
```
Comment (snuyanzin) [inline flink-java/src/test/java/org/apache/flink/api/java/utils/AbstractParameterToolTest.java:450]:
```suggestion
        assertThat(parameter.getUnrequestedParameters()).isEmpty();
```

File: flink-java/src/test/java/org/apache/flink/api/java/utils/AbstractParameterToolTest.java [hunk 4]
```diff
@@ -19,463 +19,458 @@
...
 
         // test repeated access
-        Assert.assertTrue(parameter.has("boolean"));
-        Assert.assertEquals(Collections.emptySet(), parameter.getUnrequestedParameters());
+        assertThat(parameter.has("boolean")).isTrue();
+        assertThat(parameter.getUnrequestedParameters()).isEqualTo(Collections.emptySet());
```
Comment (snuyanzin) [inline flink-java/src/test/java/org/apache/flink/api/java/utils/AbstractParameterToolTest.java:454]:
```suggestion
        assertThat(parameter.getUnrequestedParameters()).isEmpty();
```

File: flink-java/src/test/java/org/apache/flink/api/java/utils/AbstractParameterToolTest.java [hunk 5]
```diff
@@ -19,463 +19,458 @@
...
     }
 
     @Test
-    public void testUnrequestedRequired() {
+    void testUnrequestedRequired() {
         AbstractParameterTool parameter =
                 createParameterToolFromArgs(new String[] {"-required", "∞"});
-        Assert.assertEquals(createHashSet("required"), parameter.getUnrequestedParameters());
+        assertThat(parameter.getUnrequestedParameters()).isEqualTo(createHashSet("required"));
```
Comment (snuyanzin) [inline flink-java/src/test/java/org/apache/flink/api/java/utils/AbstractParameterToolTest.java:461]:
```suggestion
        assertThat(parameter.getUnrequestedParameters()).containsExactly("required");
```

File: flink-java/src/test/java/org/apache/flink/api/java/utils/AbstractParameterToolTest.java [hunk 6]
```diff
@@ -19,463 +19,458 @@
...
 
         // test parameter access
-        Assert.assertEquals("∞", parameter.getRequired("required"));
-        Assert.assertEquals(Collections.emptySet(), parameter.getUnrequestedParameters());
+        assertThat(parameter.getRequired("required")).isEqualTo("∞");
+        assertThat(parameter.getUnrequestedParameters()).isEqualTo(Collections.emptySet());
```
Comment (snuyanzin) [inline flink-java/src/test/java/org/apache/flink/api/java/utils/AbstractParameterToolTest.java:465]:
```suggestion
        assertThat(parameter.getUnrequestedParameters()).isEmpty();
```

File: flink-java/src/test/java/org/apache/flink/api/java/utils/AbstractParameterToolTest.java [hunk 7]
```diff
@@ -19,463 +19,458 @@
...
 
         // test repeated access
-        Assert.assertEquals("∞", parameter.getRequired("required"));
-        Assert.assertEquals(Collections.emptySet(), parameter.getUnrequestedParameters());
+        assertThat(parameter.getRequired("required")).isEqualTo("∞");
+        assertThat(parameter.getUnrequestedParameters()).isEqualTo(Collections.emptySet());
```
Comment (snuyanzin) [inline flink-java/src/test/java/org/apache/flink/api/java/utils/AbstractParameterToolTest.java:469]:
```suggestion
        assertThat(parameter.getUnrequestedParameters()).isEmpty();
```

File: flink-java/src/test/java/org/apache/flink/api/java/utils/MultipleParameterToolTest.java [hunk 8]
```diff
@@ -52,37 +55,38 @@ public void testFromCliArgsWithMultipleParameters() {
...
                         createParameterToolFromArgs(
                                 new String[] {"--multi", "v1", "--multi", "v2", "--multi2", "vv1"});
-        Assert.assertEquals(createHashSet("multi", "multi2"), parameter.getUnrequestedParameters());
+        assertThat(parameter.getUnrequestedParameters())
+                .isEqualTo(createHashSet("multi", "multi2"));
 
-        Assert.assertEquals(Arrays.asList("v1", "v2"), parameter.getMultiParameter("multi"));
-        Assert.assertEquals(createHashSet("multi2"), parameter.getUnrequestedParameters());
+        assertThat(parameter.getMultiParameter("multi")).isEqualTo(Arrays.asList("v1", "v2"));
+        assertThat(parameter.getUnrequestedParameters()).isEqualTo(createHashSet("multi2"));
```
Comment (snuyanzin) [inline flink-java/src/test/java/org/apache/flink/api/java/utils/MultipleParameterToolTest.java:81]:
```suggestion
        assertThat(parameter.getMultiParameter("multi")).containsExactly("v1", "v2");
        assertThat(parameter.getUnrequestedParameters()).containsExactly("multi2");
```

File: flink-java/src/test/java/org/apache/flink/api/java/utils/MultipleParameterToolTest.java [hunk 9]
```diff
@@ -52,37 +55,38 @@ public void testFromCliArgsWithMultipleParameters() {
...
 
-        Assert.assertEquals(
-                Collections.singletonList("vv1"), parameter.getMultiParameterRequired("multi2"));
-        Assert.assertEquals(Collections.emptySet(), parameter.getUnrequestedParameters());
+        assertThat(parameter.getMultiParameterRequired("multi2"))
+                .isEqualTo(Collections.singletonList("vv1"));
+        assertThat(parameter.getUnrequestedParameters()).isEqualTo(Collections.emptySet());
```
Comment (snuyanzin) [inline flink-java/src/test/java/org/apache/flink/api/java/utils/MultipleParameterToolTest.java:85]:
```suggestion
        assertThat(parameter.getMultiParameterRequired("multi2"))
                .containsExactly("vv1");
        assertThat(parameter.getUnrequestedParameters()).isEmpty();
```

File: flink-java/src/test/java/org/apache/flink/api/java/utils/MultipleParameterToolTest.java [hunk 10]
```diff
@@ -104,7 +108,7 @@ public void testMerged() {
                                 });
         MultipleParameterTool parameter = parameter1.mergeWith(parameter2);
         validate(parameter);
-        Assert.assertEquals(Arrays.asList("v1", "v2", "v3"), parameter.getMultiParameter("merge"));
+        assertThat(parameter.getMultiParameter("merge")).isEqualTo(Arrays.asList("v1", "v2", "v3"));
```
Comment (snuyanzin) [inline flink-java/src/test/java/org/apache/flink/api/java/utils/MultipleParameterToolTest.java:111]:
```suggestion
        assertThat(parameter.getMultiParameter("merge")).containsExactly("v1", "v2", "v3");
```

**Review by snuyanzin — COMMENTED** [review]
Thanks for the contribution
I made some comments

File: flink-java/src/test/java/org/apache/flink/api/java/io/CSVReaderTest.java [hunk 11]
```diff
@@ -40,258 +40,262 @@
...
+        assertThat(reader.getCharset()).isEqualTo("US-ASCII");
     }
 
     @Test
-    public void testIncludeFieldsDense() {
+    void testIncludeFieldsDense() {
         CsvReader reader = getCsvReader();
         reader.includeFields(true, true, true);
-        Assert.assertTrue(Arrays.equals(new boolean[] {true, true, true}, reader.includedMask));
+        assertThat(Arrays.equals(new boolean[] {true, true, true}, reader.includedMask)).isTrue();
```
[the remark this answers is not in the document]
Comment (RocMarshal) [reply flink-java/src/test/java/org/apache/flink/api/java/io/CSVReaderTest.java]: Good catch!

File: flink-java/src/test/java/org/apache/flink/api/java/io/RowCsvInputFormatTest.java [hunk 12]
```diff
@@ -225,22 +213,20 @@ public void ignoreMultiCharPrefixComments() throws Exception {
         Row result = new Row(3);
 
         result = format.nextRecord(result);
-        assertNotNull(result);
-        assertEquals("this is", result.getField(0));
-        assertEquals(1, result.getField(1));
-        assertEquals(2.0, result.getField(2));
+        assertThat(result.getField(0)).isEqualTo("this is");
+        assertThat(result.getField(1)).isEqualTo(1);
```
[the remark this answers is not in the document]
Comment (RocMarshal) [reply flink-java/src/test/java/org/apache/flink/api/java/io/RowCsvInputFormatTest.java]: Not applicable

File: flink-java/src/test/java/org/apache/flink/api/java/io/RowCsvInputFormatTest.java [hunk 13]
```diff
@@ -460,28 +431,26 @@ public void testIntegerFields() throws Exception {
...
-        assertEquals(666, result.getField(0));
-        assertEquals(777, result.getField(1));
-        assertEquals(888, result.getField(2));
-        assertEquals(999, result.getField(3));
-        assertEquals(0, result.getField(4));
+        assertThat(result.getField(0)).isEqualTo(666);
+        assertThat(result.getField(1)).isEqualTo(777);
+        assertThat(result.getField(2)).isEqualTo(888);
+        assertThat(result.getField(3)).isEqualTo(999);
+        assertThat(result.getField(4)).isEqualTo(0);
```
[the remark this answers is not in the document]
Comment (RocMarshal) [reply flink-java/src/test/java/org/apache/flink/api/java/io/RowCsvInputFormatTest.java]: N.A

File: flink-java/src/test/java/org/apache/flink/api/java/io/RowCsvInputFormatTest.java [hunk 14]
```diff
@@ -547,28 +516,28 @@ public void testDoubleFields() throws Exception {
...
-        assertEquals(77.7, result.getField(1));
-        assertEquals(88.8, result.getField(2));
-        assertEquals(99.9, result.getField(3));
-        assertEquals(0.0, result.getField(4));
+        assertThat(result).isNotNull();
+        assertThat(result.getField(0)).isEqualTo(66.6);
+        assertThat(result.getField(1)).isEqualTo(77.7);
+        assertThat(result.getField(2)).isEqualTo(88.8);
+        assertThat(result.getField(3)).isEqualTo(99.9);
+        assertThat(result.getField(4)).isEqualTo(0.0);
```
[the remark this answers is not in the document]
Comment (RocMarshal) [reply flink-java/src/test/java/org/apache/flink/api/java/io/RowCsvInputFormatTest.java]: N.A

File: flink-java/src/test/java/org/apache/flink/api/java/io/RowCsvInputFormatTest.java [hunk 15]
```diff
@@ -660,24 +629,24 @@ public void testReadSparseWithPositionSetter() throws Exception {
...
+        assertThat(result.getField(1)).isEqualTo(444);
+        assertThat(result.getField(2)).isEqualTo(888);
 
         result = format.nextRecord(result);
-        assertNotNull(result);
-        assertEquals(0, result.getField(0));
-        assertEquals(777, result.getField(1));
-        assertEquals(333, result.getField(2));
+        assertThat(result).isNotNull();
+        assertThat(result.getField(0)).isEqualTo(0);
```
[the remark this answers is not in the document]
Comment (RocMarshal) [reply flink-java/src/test/java/org/apache/flink/api/java/io/RowCsvInputFormatTest.java]: N.A

File: flink-java/src/test/java/org/apache/flink/api/java/io/RowCsvInputFormatTest.java [hunk 16]
```diff
@@ -699,24 +668,24 @@ public void testReadSparseWithMask() throws Exception {
...
```
[the remark this answers is not in the document]
Comment (RocMarshal) [reply flink-java/src/test/java/org/apache/flink/api/java/io/RowCsvInputFormatTest.java]: N.A

File: flink-java/src/test/java/org/apache/flink/api/java/io/RowCsvInputFormatTest.java [hunk 17]
```diff
@@ -970,21 +939,21 @@ public void testScanOrder() throws Exception {
...
         // check second row
         result = format.nextRecord(result);
-        assertNotNull(result);
-        assertEquals(333, result.getField(0));
-        assertEquals(777, result.getField(1));
-        assertEquals(0, result.getField(2));
+        assertThat(result).isNotNull();
+        assertThat(result.getField(0)).isEqualTo(333);
+        assertThat(result.getField(1)).isEqualTo(777);
+        assertThat(result.getField(2)).isEqualTo(0);
```
[the remark this answers is not in the document]
Comment (RocMarshal) [reply flink-java/src/test/java/org/apache/flink/api/java/io/RowCsvInputFormatTest.java]: N.A

File: flink-java/src/test/java/org/apache/flink/api/java/io/TextInputFormatTest.java [hunk 18]
```diff
@@ -144,23 +130,24 @@ public void testNestedFileRead() throws IOException {
...
+    private void testRemovingTrailingCR(File tmpFile, String lineBreaker, String delimiter)
+            throws IOException {
         String first = "First line";
         String second = "Second line";
         String content = first + lineBreaker + second + lineBreaker;
 
         // create input file
         File tempFile =
-                File.createTempFile("TextInputFormatTest", "tmp", temporaryFolder.getRoot());
+                File.createTempFile("TextInputFormatTest", UUID.randomUUID().toString(), tmpFile);
```
[the remark this answers is not in the document]
Comment (RocMarshal) [reply flink-java/src/test/java/org/apache/flink/api/java/io/TextInputFormatTest.java:150]: Updated.

File: flink-java/src/test/java/org/apache/flink/api/java/operator/FirstNOperatorTest.java [hunk 19]
```diff
@@ -143,34 +140,32 @@ public void testGroupedSortedFirstN() {
...
             tupleDs.groupBy(1, 3).sortGroup(4, Order.ASCENDING).first(10);
         } catch (Exception e) {
-            Assert.fail();
+            fail(e.getMessage());
         }
 
         // should not work n == 0
         try {
             tupleDs.groupBy(0).sortGroup(4, Order.ASCENDING).first(0);
-            Assert.fail();
```
[the remark this answers is not in the document]
Comment (RocMarshal) [reply flink-java/src/test/java/org/apache/flink/api/java/operator/FirstNOperatorTest.java:159]: I'll replace the block with assertThatThrownBy

File: flink-java/src/test/java/org/apache/flink/api/java/operator/MaxByOperatorTest.java [hunk 20]
```diff
@@ -65,75 +67,76 @@ public void testMaxByKeyFieldsDataset() {
         try {
             tupleDs.maxBy(4, 0, 1, 2, 3);
         } catch (Exception e) {
-            Assert.fail();
+            fail(null);
```
[the remark this answers is not in the document]
Comment (RocMarshal) [reply flink-java/src/test/java/org/apache/flink/api/java/operator/MaxByOperatorTest.java:70]: Maybe the null would be ```e.getMassage()```.
It'll be in assertThatThrownBy format coding.

**Comment by RocMarshal** [comment]
Hi, @snuyanzin Thank you very much for the comments. 
I updated based on your suggestions . PTAL.
CC @gaoyunhaii

**Comment by RocMarshal** [comment]
@flinkbot run azure

**Comment by RocMarshal** [comment]
@flinkbot run azure

**Comment by RocMarshal** [comment]
@flinkbot run azure

**Comment by gaoyunhaii** [comment]
Thanks @RocMarshal for the PR and @snuyanzin for the review! LGTM and will merge~

**Review by gaoyunhaii — APPROVED** [review]

**Pull request closed without merging**